packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -454,21 +504,29 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       }
     : null;
 
+  const expandedWidth =
+    dimensions.viewportOuterSize.width - dimensions.columnsTotalWidth - scrollbarWidth;
+  const emptyCellWidth = Math.max(0, expandedWidth);
+
   return (
     <div
       ref={handleRef}
       data-id={rowId}
       data-rowindex={index}
       role="row"
-      className={clsx(...rowClassNames, classes.root, hovered ? 'Mui-hovered' : null, className)}
+      className={clsx(...rowClassNames, classes.root, className)}
       aria-rowindex={ariaRowIndex}
       aria-selected={selected}
       style={style}
       {...eventHandlers}
       {...other}
     >
+      {leftCells}
       {cells}
       {emptyCellWidth > 0 && <EmptyCell width={emptyCellWidth} />}
+      {rightCells.length > 0 && <div role="presentation" style={{ flex: '1' }} />}
+      {rightCells}
+      <ScrollbarFiller pinnedRight={pinnedColumns.right.length > 0} />
```

Comment:
```suggestion
      {scrollbarWidth !== 0 && <ScrollbarFiller pinnedRight={pinnedColumns.right.length > 0} />}
```

No need for fillers when `scrollbarWidth === 0` - this is often the case on MacOS